### PR TITLE
new: ignore empty fields

### PR DESCRIPTION
**Pull request by qqshfox**
## Before

```console
$ echo '{"emptyString":"","zero":0,"emptyArray":[],"emptyObject":{},"false":false}' | hl -P
        ---         |(?)| emptystring='' zero=0 emptyarray=[] emptyobject={ } false=false
```

## After

```console
$ echo '{"emptyString":"","zero":0,"emptyArray":[],"emptyObject":{},"false":false}' | ./target/debug/hl -P --hide-empty-fields
        ---         |(?)| zero=0 false=false
```

### CODE REVIEW

**Comment by pamburus** [comment]
Thank you for your pull request!

But I would like not to hide empty fields by default and even without a configuration option.
It may be significant to distinguish between empty field or missing field.

Please add command-line flags for it `--hide-empty-fields` (and optionally `--show-empty-fields` to undo `--hide-empty-fields`).
If you want it by default, add an alias to the `hl` command with `--hide-empty-fields` in your environment.

**Review by pamburus — CHANGES_REQUESTED** [review]
Please add configuration option and keep current behavior by default.

**Comment by qqshfox** [comment]
> Thank you for your pull request!
> 
> But I would like not to hide empty fields by default and even without a configuration option.
> It may be significant to distinguish between empty field or missing field.
> 
> Please add command-line flags for it `--hide-empty-fields` (and optionally `--show-empty-fields` to undo `--hide-empty-fields`).
> If you want it by default, add an alias to the `hl` command with `--hide-empty-fields` in your environment.

@pamburus Thanks for the review. New flag added.

File: src/main.rs [hunk 1]
```diff
@@ -86,6 +86,14 @@ struct Opt {
     /// Files to process
     #[structopt(name = "FILE", parse(from_os_str))]
     files: Vec<PathBuf>,
+    //
+    /// Hide empty fields, only applys for null, string, object and array fields.
+    #[structopt(long, short = "H")]
+    hide_empty_fields: bool,
+    //
+    /// Show empty fields, overrides --hide_empty_fields option.
```
Comment (pamburus) [inline src/main.rs]: `structopt` automatically replaces underscores with hyphens for option names.
But it does not replace them in comments.
So, please use correct style in comments (which are automatically converted to option descriptions).
Replace `--hide_empty_fields` -> `--hide-empty-fields`

File: src/main.rs [hunk 2]
```diff
@@ -86,6 +86,14 @@ struct Opt {
     /// Files to process
     #[structopt(name = "FILE", parse(from_os_str))]
     files: Vec<PathBuf>,
+    //
+    /// Hide empty fields, only applys for null, string, object and array fields.
```
Comment (pamburus) [inline src/main.rs]: Replace `only applys for null, string, object and array fields.` -> `applies for null, string, object and array fields only.`

**Review by pamburus — CHANGES_REQUESTED** [review]

**Review by pamburus — APPROVED** [review]

**Pull request merged**